src/components/Chat/ChatBox.tsx

```diff
@@ -1,69 +1,132 @@
+    resetInput();
+  }, [message]);
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [chatHistory, state]);
 
-  // Focus the user on the input when state changes
-  const inputRef = useRef<HTMLInputElement>(null);
   const focusInput = () => {
     inputRef.current?.focus();
   };
 
-  useEffect(() => {
-    focusInput();
-  }, [state]);
+  const resetInput = () => {
+    if (inputRef.current) {
+      inputRef.current.style.height = "auto";
+      inputRef.current.style.height = `${inputRef.current.scrollHeight}px`;
+    }
+  };
+
+  const scrollToBottom = () => {
+    bottomRef.current?.scrollIntoView({ behavior: "smooth" });
+  };
 
   const handleFormSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
+
+    if (state === "loading") {
+      return;
+    }
+
     const newMessage = message.trim();
     setMessage("");
     if (newMessage !== "") {
+      scrollToBottom();
       try {
-        await sendMessage(newMessage);
+        sendMessage(newMessage);
       } catch (err) {
         logger.error(`Error while sending message: ${err}`);
       }
     }
   };
 
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setMessage(e.target.value);
+  };
+
+  const handleKeyDown = async (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === "Enter" && !e.shiftKey) {
+      await handleFormSubmit(e);
+    }
+  };
+
+  const loading = chatHistory.at(-1)?.role === "human" && state === "loading";
+
   return (
-    <>
-      <Container className="overflow-y-auto flex-grow">
-        <Row className="flex flex-col">
-          {chatHistory.length === 0 ? (
-            <p>
-              Welcome to <b>eras</b>! Share any concerns, feelings or ideas
-              about money you have with us.
-            </p>
-          ) : (
-            chatHistory.map((chat, i) => <ChatMessage key={i} message={chat} />)
-          )}
-        </Row>
-      </Container>
-      <Row className="mt-3">
-        <Form onSubmit={handleFormSubmit} role="form">
-          <Stack direction="horizontal" gap={2}>
-            <Form.Control
-              className="me-auto"
-              placeholder="Send a message"
-              value={message}
-              onChange={(e) => setMessage(e.target.value)}
-            />
-            <Button className="w-25" type="submit" variant="secondary">
-              Send
-            </Button>
-          </Stack>
-        </Form>
+    <Container className="d-flex flex-column">
+      <div className={`${styles.chatBox}`}>
+        {chatHistory.length === 0 ? (
+          <p>
+            Welcome to <b>eras</b>! Share any concerns, feelings or ideas about
+            money you have with us.
+          </p>
+        ) : (
+          chatHistory.map((chat, i) => <ChatMessage key={i} message={chat} />)
+        )}
+        {loading ? (
+          <Row className="justify-content-center my-3">
+            <Col className="text-center">
+              <Spinner
```

Comment:
🙏 Not that I'm consistent about this, but thanks for looking out for accessibility here!

Reply:
Also, the choice and style of spinner is 😍